src/indexnotation/planarordering.jl

```diff
@@ -0,0 +1,164 @@
+# Functions for ordering contraction indices so as to avoid unnecessary braiding.
+# TODO Maybe this should be a submodule? It would only need to export parsegraph and
+# nextpair.
+
+# A Node of the contraction graph is a Tuple of Any elements, that are the names of the
+# edges connected to that Node, in counter clockwise order.
+Node = Tuple
+# An Edge is a Tuple of Symbol that are the names of Nodes that the Edge connects. The
+# alternatively, the second element of the Tuple maybe be nothing, to indicate an open
+# index.
+Edge = Union{Tuple{Symbol, Symbol}, Tuple{Symbol, Nothing}}
+
+# TODO I just made up the name RigidGraph, there must be an established math term for this.
+"""
+A RigidGraph is a graph for which the edges connected to a vertex come with an planar
+ordering, counter clockwise around the vertex.
+"""
+struct RigidGraph
```

Comment:
I think this is called a two-dimensional [combinatorial map](https://en.wikipedia.org/wiki/Combinatorial_map) .

Reply:
Also related to [https://en.wikipedia.org/wiki/Rotation_system] , not sure what the difference is or what the terminology entails specifically.

Reply:
Thanks. Should rename that then.